examples/minimal/packages/client-react-external-wallet/src/App.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, useEffect } from "react";
+import { BaseError } from "viem";
+import {
+  usePublicClient,
+  useWalletClient,
+  useAccount,
+  useConnect,
+  useDisconnect,
+  useNetwork,
+  useSwitchNetwork,
+} from "wagmi";
+import { share } from "rxjs";
+import { useNetworkConfig, useWorldContract } from "./mud/networkConfig";
+import { syncStore, StoreProvider, useStore, type Store } from "./mud/store";
+import mudConfig from "contracts/mud.config";
+
+export const App = () => {
+  const networkConfig = useNetworkConfig();
+  const publicClient = usePublicClient();
+  const { isConnected } = useAccount();
+
+  const [store, setStore] = useState<Store | null>(null);
+
+  useEffect(() => {
+    syncStore(networkConfig, publicClient).then(setStore);
+  }, []);
+
+  return (
+    <>
+      <Connect />
+      {isConnected && <Wallet />}
+      {store && (
+        <StoreProvider store={store}>
+          <Synced />
+        </StoreProvider>
+      )}
+    </>
+  );
+};
```

Comment:
The way these various parts are separated is a little hard for me to grok. I wonder if there are other approaches that would make this easier to understand.

One pattern we've used in some apps is a [loading screen that is shown while we're syncing data](https://github.com/latticexyz/emojimon/blob/complete/packages/client/src/App.tsx#L6-L29)  from the chain or indexer. We could use that here while waiting for a public client.

And like mentioned above, we could separate the things that depend on the wallet client so it doesn't block showing the UI, etc.